### PR TITLE
feat: scheduling groundwork — shared authorization, pagination, and queue trigger path

### PR DESCRIPTION
**Pull request by induwara-yaala**
## Description
Groundwork for the scheduling capability (#629), split out ahead of the feature itself. No
scheduling behavior ships here: this PR lands the spec set, extracts the pieces the scheduling
package will reuse, and prepares the queue path to consume scheduled triggers. Phases 0, 1 and 2
of `docs/specs/629-scheduled-tasks/plan.md`.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [x] Test update
- [ ] CI/CD update
- [ ] Other (please describe):

## Changes Made

Groundwork for the scheduling capability (#629), split out ahead of the feature itself. No
scheduling behavior ships here: this PR lands the spec set, extracts the pieces the scheduling
package will reuse, and prepares the queue path to consume scheduled triggers. Phases 0, 1 and 2
of `docs/specs/629-scheduled-tasks/plan.md`.

## Testing

- [x] Unit tests pass locally
- [x] Integration tests pass locally
- [x] Manual testing completed
- [x] New tests added for changes

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: ak-py/src/agentkernel/core/chat_service.py [hunk 1]
```diff
@@ -412,6 +420,22 @@ def _stream() -> Generator[StreamChunk, None, None]:
...
+        run can attribute work to the caller. The cache is volatile: ``Runtime.run``/``stream``
+        clear it when the run ends, so the key never leaks into the next run of the same session.
+
+        :param handler: The initialized agent handler owning the session
+        :param req: The request whose user_id is published (a request without one publishes nothing)
+        :return: None
+        """
+        if not req.user_id or not handler.service or not handler.service.session:
+            return
+        handler.service.session.get_volatile_cache().set(ACTING_USER_CACHE_KEY, req.user_id)
```
Comment (amithad) [inline ak-py/src/agentkernel/core/chat_service.py]: **[suggestion]** Acting-user cache write happens before the per-session lock, so it can race across concurrent requests on the same `session_id`.

- Why: `_propagate_acting_user` (called at lines 353/368/387/412, before `handler.run_async`/`run_sync`/etc.) writes directly to `handler.service.session.get_volatile_cache()` outside of `Runtime.run()`'s `async with session:` lock (`core/runtime.py:199`). For the default `InMemorySessionStore`, `load()` returns the *same* `Session` object by reference for a given `session_id` (`core/session/in_memory.py:23-27`), and `Runtime.run()`'s `finally` clears the *entire* volatile cache (`core/runtime.py:221-222`) once a run completes.
- Failure scenario: two concurrent `ChatService.execute()` (or `execute_sync`/`execute_stream`/`execute_stream_sync`) calls for the same `session_id` (double-submit, retry, multi-tab) each call `_propagate_acting_user` before either enters the lock. Whichever write lands last wins the shared cache; the first request to actually acquire the lock and run may then see the *other* request's `user_id` inside its hooks/tools. When that first run finishes, its `finally` clears the cache, so the second request — whose own `user_id` was already overwritten/cleared — sees `None` once it runs.
- This is dormant today (nothing in this PR reads `ACTING_USER_CACHE_KEY` outside the test), but `design.md`/`spec.md` for #629 plan to use this exact value for schedule-tool *ownership* in Phase 4 — a misattributed acting user there means a schedule created under the wrong owner.
- Suggestion: move the acting-user publish inside `Runtime.run()`/`stream()` (under the session lock), or otherwise guarantee it's read-and-cleared atomically with the run it belongs to, before Phase 4 tools start depending on it for authorization decisions.

**Review by amithad — COMMENTED** [review]
**Overall:** Solid, closely-spec'd groundwork PR. This PR carries three spec documents (`design.md`, `spec.md`, `plan.md`, plus `research/`) for issue #629 and implements Phases 0-2 of `plan.md` (spec set + shared auth/pagination refactor + queue-path groundwork). I reviewed the spec set first, then checked every Phase 1/2 file against it.

**Spec review:** `design.md` is properly point-form/hierarchical/complete, and `spec.md` traces every design requirement into a concrete implementation section with verifiable `path:line` citations. I spot-checked several citations against `develop` (e.g. the `known_fields` set at `chat_service.py:126`, the four execution-core entry-point line numbers) and they check out exactly. The staged process was followed correctly (`design.md` → `spec.md` → `plan.md`, `research/` shipped alongside `design.md`).

**Implementation vs. plan.md Phases 1-2:** verified line-by-line against every iteration (1.1-1.4, 2.1-2.4). All hold up:
- `Authoriser` relocation (1.1) is a verbatim move with all five listed consumers migrated in this PR, and a guard test asserting the thread package no longer exposes it.
- `AuthorisedRESTRequestHandler` extraction (1.2) preserves the three pinned 401 detail strings verbatim.
- Pagination extraction (1.3) is a verbatim lift into `core/util/pagination.py`, with `manager.py` delegating and its now-unused `base64` import dropped.
- Queue-path groundwork (2.1-2.4): `_resolve_request_metadata`/`_get_record_attributes` body-fallback lands identically across the pipeline, ECS, and serverless runners with matching attribute precedence; ECS status-code forwarding and `_build_sync_response`'s 2xx-non-200 handling match the spec exactly (confirmed via the `RequestHandler._build_sync_response is RestHandler._build_sync_response` identity test); `ACTING_USER_CACHE_KEY` propagation is wired into all four `ChatService` entry points as specified.

Test coverage per iteration is thorough (including edge cases like malformed-body-doesn't-mask-missing-request_id, attribute-precedence-over-body, and volatile-cache-doesn't-leak-into-next-run). CI is green across the board.

**Findings:** 1 suggestion (inline) + 2 non-anchorable notes below. No blockers.

- `[suggestion]` `core/chat_service.py:437` — acting-user cache write races with the per-session lock on concurrent same-session requests (see inline comment).
- `[suggestion]` `plan.md`'s own Iteration 1.4 commits to adding the breaking `Authoriser` import change to release notes (following the `docs/specs/612-chatservice-refactor/release-notes.md` precedent), but no such file ships in this PR.
- `[nit]` The five new test files added in this PR (`test_authoriser_shared.py`, `test_ecs_agent_runner_schedule.py`, `test_ecs_output_consumer_status.py`, `test_pipeline_agent_runner_schedule.py`, `test_serverless_agent_runner_schedule.py`) aren't reflected in `.agents/skills/ak-dev-testing-conventions/SKILL.md`'s test-file inventory table (sibling files like `test_pipeline_agent_runner.py` are listed there). `plan.md`'s Phase 6 sync step only lists the later `test_schedule_*` files, so this gap isn't tracked anywhere.

No existing review comments/threads were found on this PR to dedupe against.

File: docs/specs/629-scheduled-tasks/spec.md [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+            SystemTool(name="get_schedule", description="", func=get_schedule),
+            SystemTool(name="update_schedule", description="", func=update_schedule),
+            SystemTool(name="delete_schedule", description="", func=delete_schedule)]
+```
+
+- Signatures: `create_schedule(prompt, cron=None, at=None, timezone="UTC", session_mode="reuse", agent=None)`; `update_schedule(task_id, prompt, cron=None, at=None, timezone="UTC", session_mode="reuse", status="active")` (PUT semantics); `list_schedules()`, `get_schedule(task_id)`, `delete_schedule(task_id)`.
+- Every tool starts `manager = ScheduleManager.get(); if manager is None: return _DISABLED` (`_DISABLED = json.dumps({"error": "scheduling capability is disabled"})`, the sandbox first-line pattern `sandbox/tools.py:79-81`).
+- **Acting user**: `Session.current().get_volatile_cache().get(ACTING_USER_CACHE_KEY)` (imported from `core.chat_service`). Every tool requires it: absent → `{"error": "scheduling requires a user identity: include user_id on the chat request"}`. `list_schedules` is scoped to the acting user; `get`/`update`/`delete` pass it for ownership enforcement (a `PermissionError` becomes an error JSON).
```
Comment (amithad) [inline docs/specs/629-scheduled-tasks/spec.md]: **[suggestion]** This section (and lines 105, 128-130, 135 above) describes the acting-user mechanism as it stood before this PR's last fix — it's now stale.

- Why: the implementation was corrected (in response to earlier review feedback) to close a concurrency race — `ACTING_USER_CACHE_KEY` now lives in `core/runtime.py`, not `core.chat_service`, and it's published via an `acting_user_id` parameter threaded through `AgentService`/`AgentHandler` down to `Runtime.run`/`stream`, set *inside* the per-session lock rather than by a `ChatService._propagate_acting_user` staticmethod called beforehand.
- Concretely wrong now: "imported from `core.chat_service`" (this line), the `_propagate_acting_user` staticmethod sketch (lines 127-130), and "inserted after `handler.initialize(...)`" (line 135).
- Suggestion: update this section before Phase 4 implements the schedule tools that read this key — otherwise a future PR will copy the stale import path (`from ..core.chat_service import ACTING_USER_CACHE_KEY`), which no longer resolves.

**Review by amithad — COMMENTED** [review]
**Re-review after the two follow-up commits (`2c5fbeb3`, `2c74d912`).** Both outstanding findings from the prior rounds are fixed correctly:

- Copilot's stale-`MAX_PAGE_SIZE`-docstring comment on `integration/thread/manager.py` — fixed; the docstrings now say `core.util.pagination.MAX_PAGE_SIZE`.
- My prior `[suggestion]` about the acting-user cache racing the per-session lock — fixed properly, not just patched over: `ACTING_USER_CACHE_KEY` moved to `core/runtime.py`, and `acting_user_id` is now threaded as an explicit parameter through `ChatService.execute*` → `AgentHandler.run_*` → `AgentService.run_multi`/`stream_multi` → `Runtime.run`/`stream`, where it's set *inside* `async with session:` (`core/runtime.py:207-208`, `250-252`) — the same lock that clears the volatile cache in `finally`. That closes the race I described: set and clear are now serialized by the same lock. All four `ChatService` entry points and their fakes/tests were updated consistently, and `core/__init__.py` re-exports the constant from its new home. CI is green (lint, CodeQL, unit tests, all e2e).

One new finding from this refactor, not merge-blocking but worth doing before Phase 3 starts:

- `[suggestion]` `docs/specs/629-scheduled-tasks/spec.md:379` — the spec's "ChatService interception" section (lines 105, 128-130, 135) and this line now describe a superseded design (a `ChatService._propagate_acting_user` staticmethod, `ACTING_USER_CACHE_KEY` "imported from `core.chat_service`", set *after* `handler.initialize(...)`). The shipped implementation is architecturally different in a way that matters for Phase 3/4: the constant now lives in `core.runtime`, and propagation is threaded as an `acting_user_id` parameter set inside `Runtime.run`/`stream`'s session lock rather than poked into the cache from `ChatService` beforehand. Since spec.md line 379 is exactly what a Phase 4 schedule-tools implementer would copy as an import path, it should be corrected now so it doesn't get built against a stale reference.

No other regressions found; the rest of Phases 1-2 (Authoriser relocation, `AuthorisedRESTRequestHandler`, pagination extraction, queue-path body-fallback and status forwarding) is unchanged since my last pass and still holds up against the spec.

**Review by amithad — COMMENTED** [review]
**Re-review after commit `9395c3e7`** ("update SKILL.md and plan.md to reflect relocation of Authoriser and new test files") — docs/skills-only, no source changes.

This closes out every outstanding item from both prior rounds, including the two lower-severity notes I said weren't merge-blocking:

- `docs/specs/629-scheduled-tasks/spec.md` — the "ChatService interception" section now correctly describes the shipped design (no more stale `_propagate_acting_user`/`core.chat_service` references), and a new "Acting-user propagation (`core/runtime.py`) — shipped in Phase 2" section accurately documents the fixed mechanism end to end, matching the code exactly (constant location, parameter threading through `AgentHandler`/`AgentService`/`Runtime`, set-and-clear serialized by the per-session lock). Behavioural change 9 updated to match.
- `docs/specs/629-scheduled-tasks/plan.md` — Iteration 1.4 and Phase 2's intro now call out syncing `ak-dev-testing-conventions/SKILL.md`, and Iteration 2.4's file list/steps reflect the real `core/runtime.py` + `core/service.py` + `core/chat_service.py` shape.
- `docs/specs/629-scheduled-tasks/release-notes.md` — new, accurate, and thorough: covers the `Authoriser` breaking import change, the ECS status-code parity fix, and all the Phase 1/2 improvements (`AuthValidatorAuthoriser`, `AuthorisedRESTRequestHandler`, shared pagination, queue body-fallback, acting-user propagation) with correct code samples.
- `.agents/skills/ak-dev-testing-conventions/SKILL.md` — all five new test files from this PR now have inventory rows, sensibly grouped next to their topical siblings.

I spot-checked the updated spec.md/release-notes.md claims against the actual code from my last pass and they all hold up (e.g. `from agentkernel.core import ACTING_USER_CACHE_KEY` does resolve via `core/__init__.py`'s re-export).

CI is green (52 passing checks, only the expected `update-lock-files` skip).

**No new findings.** Nothing outstanding from my side — this looks ready for a maintainer decision.

**Review by amithad — APPROVED** [review]
